### PR TITLE
fix to get lusca work with client-session library

### PR DESCRIPTION
**Pull request by dearsmk**

### CODE REVIEW

File: lib/token.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ var LENGTH = 10;
 
 function create(req, secretKey) {
 
-    var session = req.session;
+    var session = req.session || req.cookies;
     if (session === undefined) {
-        throw new Error('lusca requires req.session to be available in order to maintain state');
+        throw new Error('lusca requires either req.session or req.cookies to be available in order to maintain state');
```
Comment (thefourtheye) [inline lib/token.js]: Didn't the longer complain about line length?

**Comment by jasisk** [comment]
Closing as insufficient. Cookies are user provided. We cannot use them as a means of trusted storage of things such as the secret used to generate csrf tokens.

**Comment by jasisk** [comment]
I'll explain a little bit here.

So `client-session` works by encrypting the contents of a cookie with a known secret. That ensures that the client cannot mutate the contents of the cookie since they cannot decrypt the payload. So that gives you a way to store secure data in an insecure environment, and validate the contents the next time it comes around.

This, by itself, is susceptible to csrf.

So you still need secret and a token. You could store the secret inside the "client-session" and use that for generation and validation but there's a problem there, too. This is now susceptible to replay.

So now you'd probably want to invalidate the secret in the space between validation and new token generation or, at least leverage time-based invalidation.

Said another way: we cannot simply say, "use a session or, in place of that, cookies," as cookies are inherently insecure. If, however, there is some mechanism for securing them (like client-session), that's fine, however the data needs to be accessed through that secure interface.

In short, if you leverage the "cookieName" configuration of client-session to make it _look_ like a traditional express session, you'll be good to go. `app.use(require('client-session')({ cookieName: 'session' })`. Just be sure to add sane expiration as well.

File: lib/token.js (hunk 1 above)
Comment (jasisk) [inline lib/token.js]: Nope. There's a test failure.

**Comment by dearsmk** [comment]
Thanks!! agree with your points. However our apps uses different session names and we don't have req.session available. Hence, before calling lusca, copying actual session object to req.session, and that does the trick for me.

// lusca needs req.session to be present, before issuing _csrf cookie
  app.use( function(req, res, next ){
    var session_name = getConfigValue('session_name');
    req.session = req.session || req[session_name] || {};
    next();
  });
